### ISSUE TITLE
TCH003 false-positive when using dataclasses.KW_ONLY

### ISSUE THREAD

**Issue by flying-sheep** [astral-sh/ruff#12859]
Seen in Ruff 0.5.7

In the following code, `from __future__ import annotations` causes the `KW_ONLY` to become a string.
The dataclasses module can deal with that, but only if the object isn’t in a `if TYPE_CHECKING` block.
Ruff supports this for things like `ClassVar`¹, i.e. if a `ClassVar` is used in a dataclass, Ruff knows the ` from typing import ClassVar` needs to come out of the `if TYPE_CHECKING` block. However the same doesn’t seem to be true here.

```py
from __future__ import annotations

from dataclasses import KW_ONLY, dataclass

@dataclass
class Test:
    a: int
    _: KW_ONLY
    b: str
```

----
¹This is probably the place that should handle KW_ONLY
https://github.com/astral-sh/ruff/blob/82a3e69b8a096b0cfceae71121ca5201e28cd192/crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs#L108-L139

**Comment by charliermarsh** [comment]
Wow TIL, I’ve never seen that feature before.

**Comment by charliermarsh** [comment]
Ah, this requires:

```toml
[tool.ruff.flake8-type-checking]
strict = true
```

So probably somewhat rare (but should be fixed).

**Comment by flying-sheep** [comment]
Thank you!

**Comment by flying-sheep** [comment]
Btw.: I always do

```toml
[tool.ruff.lint.flake8-type-checking]
exempt-modules = []
strict = true
```

because that way I can use new `typing` features without having to pay attention that they are imported from `typing_extensions` or in a `if TYPE_CHECKING` block for backwards compatibility. They just always are in there.
